arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -1,64 +1,160 @@
+Once the ticket is successfully created, the two following conditions are checked: (1) if the user's L2 balance is greater than (or equal to) `maxFeePerGas * gasLimit` **and** (2) if the `maxFeePerGas` (provided by the user in the ticket submission process) is greater than (or equal to) the `l2Basefee`.
+If these conditions are both met, ticket's submission is followed by an attempt to execute it on L2 (i.e., an **auto-redeem** using the supplied gas, as if the [`redeem`](./precompiles#ArbRetryableTx) method of the ArbRetryableTx precompile had been called). 
+Depending on how much gas the sender has provided in step 1, ticket's redemption can either (1) immediately succeed or (2) fail. We explain both situations here:
+
+  - If the ticket is sucessfully auto-redeemed, it will execute with the sender, destination, callvalue, and calldata of the original submission. The submission fee is refunded to the user on L2 (`excessFeeRefundAddress`). Note that to ensure successful auto-redeem of the ticket, one could use the Arbitrum SDK which provides a [convenience function](https://github.com/OffchainLabs/arbitrum-sdk/blob/4cedb1fcf1c7302a4c3d0f8e75fb33d82bc8338d/src/lib/message/L1ToL2MessageGasEstimator.ts#L215) that returns the desired gas parameters when sending L1-L2 messages.
+
+  - If a redeem is not done at submission or the submission's initial redeem fails (for example, because the L2 gas price has increased unexpectedly), the submission fee is collected on L2 to cover the resources required to temporarily keep the ticket in memory for a fixed period (one week).
+
+
+<MermaidWithHtml>
+  <Nodes title="Automatic Redemption of the Ticket">
+    <Node id="1">Auto-redeem succeeds?</Node>
+    <Node id="2">Ticket is executed</Node>
+    <Node id="3">Ticket is deleted</Node>
+    <Node id="4">callValueRefundAddress gets refunded</Node>
+    <Connection from="1" to="2" label="yes"/>
+    <Connection from="2" to="3"/>
+    <Connection from="1" to="4" label="no"/>
+  </Nodes>
+  <NodeDescriptions>
+    <NodeDescription for='1'><code>Does the auto-redeem succeed?</code> Logic that determines if the user's L2 Balance is greater than (or equal to) <code>maxFeePerGas * gasLimit</code> && <code>maxFeePerGas</code> is greater than (or equal to) the <code>l2Basefee</code></NodeDescription>
+    <NodeDescription for='2'><code>Ticket is executed</code> Ticket is executed, the actual <code>submissionFee</code> is refunded to the  <code>excessFeeRefundAddress</code> since the ticket was not kept in the buffer on L2</NodeDescription>
+    <NodeDescription for='3'><code>Ticket is deleted</code> Ticket gets deleted from the L2 retryable buffer</NodeDescription>
+    <NodeDescription for='4'><code>callValueRefundAddress gets refunded</code> <code>callValueRefundAddress</code> gets refunded with <code>(maxGas - gasUsed) * gasPrice</code>. Note that this amount is capped by <code>l1CallValue</code> in the auto-redeem</NodeDescription>
```

Comment:
Should we add one more step indicating that the Ticket stays created for one week if not autoredeemed?
(Not sure about this)